testing/mock.ts

```diff
@@ -643,15 +654,15 @@ function methodSpy<
   Return,
 >(self: Self, property: keyof Self): MethodSpy<Self, Args, Return> {
   if (typeof self[property] !== "function") {
-    throw new MockError("Property is not an instance method");
+    throw new TypeError("Property is not an instance method");
   }
   if (isSpy(self[property])) {
     throw new MockError("Already spying on instance method");
   }
 
   const propertyDescriptor = Object.getOwnPropertyDescriptor(self, property);
   if (propertyDescriptor && !propertyDescriptor.configurable) {
-    throw new MockError("Cannot spy on non-configurable instance method");
+    throw new TypeError("Cannot spy on non-configurable instance method");
```

Comment:
Similarly, why isn't this MockError?

Reply:
Actually, yes, this could be reverted.

Reply:
[`TypeError`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) documentation:
> A `TypeError` may be thrown when:
> - an operand or argument passed to a function is incompatible with the type expected by that operator or function; or
> - when attempting to modify a value that cannot be changed; or
> - when attempting to use a value in an inappropriate way.